### PR TITLE
[process] Fix Process.Exited event triggered too soon

### PR DESCRIPTION
**Pull request by ludovic-henry**

### CODE REVIEW

**Comment by akoeplinger** [comment]
Can you please add a bit more context about what the problem was to the commit message? In which commit was it introduced?

File: mcs/class/System/Test/System.Diagnostics/ProcessTest.cs [hunk 1]
```diff
@@ -1085,5 +1085,31 @@ public void TestDoubleBeginErrorReadLine ()
 			}
 		}
 #endif // MONO_FEATURE_PROCESS_START
+
+#if MONO_FEATURE_PROCESS_START
```
Comment (akoeplinger) [inline mcs/class/System/Test/System.Diagnostics/ProcessTest.cs]: nit: no need to start a new #if section, combine it with the above

File: mcs/class/System/Test/System.Diagnostics/ProcessTest.cs [hunk 2]
```diff
@@ -1085,5 +1085,31 @@ public void TestDoubleBeginErrorReadLine ()
...
+			}
+
+			var sleeptime = 5;
+			var mre = new ManualResetEvent (false);
+			var process = Process.Start("sleep", sleeptime.ToString ());
+			process.EnableRaisingEvents = true;
+			process.Exited += (sender, e) => {
+				mre.Set ();
+			};
+			if (mre.WaitOne ((sleeptime - 2) * 1000))
```
Comment (marek-safar) [inline mcs/class/System/Test/System.Diagnostics/ProcessTest.cs]: Assert.IsTrue instead

File: mcs/class/System/Test/System.Diagnostics/ProcessTest.cs [hunk 3]
```diff
@@ -1085,5 +1085,31 @@ public void TestDoubleBeginErrorReadLine ()
...
+			case PlatformID.MacOSX:
+				break;
+			default:
+				Assert.Ignore ("no sleep command on windows");
+				return;
+			}
+
+			var sleeptime = 5;
+			var mre = new ManualResetEvent (false);
+			var process = Process.Start("sleep", sleeptime.ToString ());
```
Comment (marek-safar) [inline mcs/class/System/Test/System.Diagnostics/ProcessTest.cs]: I think we can generic test utility to launch any available external process somewhere

**Comment by akoeplinger** [comment]
@ludovic-henry thanks for the update, but it doesn't really address _why_ we were seeing this just recently.

From looking at the code, it's because in https://github.com/mono/mono/commit/98284703dec53f1dd5664bfb25ee589d7aef430f the function `mono_os_cond_timedwait` was changed to return -1 instead of ETIMEDOUT and the call site wasn't updated, correct? Do we have other call sites that need to be updated?

**Pull request merged**